plugins/com.contrastsecurity.ide.eclipse.ui/src/com/contrastsecurity/ide/eclipse/ui/internal/views/FilterDialog.java

```diff
@@ -0,0 +1,714 @@
+			updateApplicationCombo(orgUuid, true, applications);
+		}
+	};
+
+	private ISelectionChangedListener listener = new ISelectionChangedListener() {
+		@Override
+		public void selectionChanged(SelectionChangedEvent event) {
+
+			final String selection = (String) ((IStructuredSelection) lastDetectedCombo.getSelection())
+					.getFirstElement();
+
+			if (!selection.equals((Constants.LAST_DETECTED_CUSTOM))) {
+				dateTimeFrom.setEnabled(false);
+				dateTimeTo.setEnabled(false);
+			}
+			LocalDateTime localDateTime = LocalDateTime.now();
+			switch (selection) {
+			case Constants.LAST_DETECTED_ALL:
+				prefs.remove(Constants.LAST_DETECTED_FROM);
+				prefs.remove(Constants.LAST_DETECTED_TO);
+				break;
+			case Constants.LAST_DETECTED_HOUR:
+				LocalDateTime localDateTimeMinusHour = localDateTime.minusHours(1);
+
+				dateTimeFrom.setYear(localDateTimeMinusHour.getYear());
+				dateTimeFrom.setDay(localDateTimeMinusHour.getDayOfMonth());
+				dateTimeFrom.setMonth(localDateTimeMinusHour.getMonthValue() - 1);
+				dateTimeFrom.setHours(localDateTimeMinusHour.getHour());
+				dateTimeFrom.setMinutes(localDateTimeMinusHour.getMinute());
+				dateTimeFrom.setSeconds(localDateTimeMinusHour.getSecond());
+
+				break;
+			case Constants.LAST_DETECTED_DAY:
+				LocalDateTime localDateTimeMinusDay = localDateTime.minusDays(1);
+
+				dateTimeFrom.setYear(localDateTimeMinusDay.getYear());
+				dateTimeFrom.setDay(localDateTimeMinusDay.getDayOfMonth());
+				dateTimeFrom.setMonth(localDateTimeMinusDay.getMonthValue() - 1);
+				dateTimeFrom.setHours(localDateTimeMinusDay.getHour());
+				dateTimeFrom.setMinutes(localDateTimeMinusDay.getMinute());
+				dateTimeFrom.setSeconds(localDateTimeMinusDay.getSecond());
+				break;
+			case Constants.LAST_DETECTED_WEEK:
+				LocalDateTime localDateTimeMinusWeek = localDateTime.minusWeeks(1);
+
+				dateTimeFrom.setYear(localDateTimeMinusWeek.getYear());
+				dateTimeFrom.setDay(localDateTimeMinusWeek.getDayOfMonth());
+				dateTimeFrom.setMonth(localDateTimeMinusWeek.getMonthValue() - 1);
+				dateTimeFrom.setHours(localDateTimeMinusWeek.getHour());
+				dateTimeFrom.setMinutes(localDateTimeMinusWeek.getMinute());
+				dateTimeFrom.setSeconds(localDateTimeMinusWeek.getSecond());
+				break;
+			case Constants.LAST_DETECTED_MONTH:
+				LocalDateTime localDateTimeMinusMonth = localDateTime.minusMonths(1);
+
+				dateTimeFrom.setYear(localDateTimeMinusMonth.getYear());
+				dateTimeFrom.setDay(localDateTimeMinusMonth.getDayOfMonth());
+				dateTimeFrom.setMonth(localDateTimeMinusMonth.getMonthValue() - 1);
+				dateTimeFrom.setHours(localDateTimeMinusMonth.getHour());
+				dateTimeFrom.setMinutes(localDateTimeMinusMonth.getMinute());
+				dateTimeFrom.setSeconds(localDateTimeMinusMonth.getSecond());
+				break;
+			case Constants.LAST_DETECTED_YEAR:
+				LocalDateTime localDateTimeMinusYear = localDateTime.minusYears(1);
+
+				dateTimeFrom.setYear(localDateTimeMinusYear.getYear());
```

Comment:
Create a function to fill the dateTimeFrom object and pass the LocalDateTime as parameter to avoid repeating code.

```
private void fillDateTimeFrom(LocalDateTime date) {
    dateTimeFrom.setYear(date.getYear());
    //And so on...
}
```

Reply:
Done.